pkg/storage/storage.go

```diff
@@ -181,3 +184,48 @@ func parseCommit(r io.Reader, hash string) (Commit, error) {
 
 	return c, nil
 }
+
+//TreeEntry is a file or folder at a given path in a repository
+type TreeEntry struct {
+	Mode   string
+	Type   string
+	Object string
+	Path   string
+}
+
+//Tree returns the files and folders at a given rev at a path in a repository
+func (s *storage) Tree(ctx context.Context, owner string, name string, rev string, path string) ([]TreeEntry, error) {
+	args := []string{"ls-tree", rev, path}
+	cmd := exec.CommandContext(ctx, s.git, args...)
+	cmd.Dir = filepath.Join(s.root, owner, name)
+	out, err := cmd.Output()
```

Comment:
This is unbounded output... huuge directories could cause OOM-errors :thinking:

Reply:
Good point. Should we try to do this streaming then and add a io.LimitReader, or how would you solve this?

Reply:
I'd use `LimitReader` yes, but then you'd have to handle (drop?) partial reads, and return an overflow-boolean :thinking: And somehow you'd still wanna paginate :thinking:

On second thought. I'd add a TODO noting that the output can be huge and that pagination would be nice, but I'd still try to stream it :slightly_smiling_face: